### PR TITLE
Added JS Roadmap

### PR DESCRIPTION
**Pull request by dhairyagothi**
## Description
Added Javascript roadmap 
[Please include a brief description of the changes or features added]

## Type of PR

- [ ] Bug fix
- [ ] Feature enhancement
- [X] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [X] I have performed a self-review of my code.
- [X] I have read and followed the Contribution Guidelines.
- [X] I have tested the changes thoroughly before submitting this pull request.
- [X] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have followed the code style guidelines of this project.
- [X] I have checked for any existing open issues that my pull request may address.
- [X] I have ensured that my changes do not break any existing functionality.
- [X] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [X] I have read the resources for guidance listed below.
- [X] I have followed security best practices in my code changes.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Please read the following resources before submitting your contribution:

- [ ] [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [ ] [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [ ] [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [ ] [Markdown Guide](https://www.markdownguide.org/)
- [ ] [MDX Documentation](https://mdxjs.com/docs/)
- [ ] [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- [ ] [Tabs: Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [ ] [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [ ] [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex) 

## Contribution Guidelines

Thank you for considering contributing to our project! To ensure smooth collaboration and effective contribution management, please adhere to the following guidelines:

### Issue Creation

1. **Limit on Issues:**
   - Each contributor is allowed to create a maximum of **4 issues per day**. This helps us manage and address issues efficiently.

### Contribution Levels

2. **Basic Contributions:**
   - This project is primarily focused on documentation. Most of the setup has been completed, so contributors will generally need to work on basic code tasks, such as writing tests.
   - For these tasks, issues will be assigned the **level1** label.

3. **Acknowledging Hard Work:**
   - If a contributor puts in significant effort on a task, the issue will be upgraded to **level2**. This is our way of recognizing and appreciating extra effort.

4. **Feature Additions and Component Work:**
   - Contributors working on new features or components using JSX/TSX will be assigned a level based on the complexity and quality of their work.
   - The more complex and valuable the contribution, the higher the level assigned.

### Level Definitions

- **Level1:**
  - Tasks are straightforward, such as fixing minor bugs, writing tests, or making simple documentation updates.
- **Level2:**
  - Tasks require more effort, such as addressing complex bugs, improving existing features, or making substantial documentation improvements.
- **Level3:**
  - Tasks are highly complex and involve significant new feature development, major refactoring, or extensive contributions to the project’s core components.

We look forward to your contributions and appreciate your effort in helping us improve the project!

### CODE REVIEW

File: src/components/Roadmap/index.tsx [hunk 1]
```diff
@@ -14,8 +14,8 @@ const techCategories: TechCategory[] = [
     title: "Frontend",
     technologies: [
       { id: 1, name: "HTML", link: "/roadmap/html" },
-      { id: 2, name: "CSS", link: "#" },
-      { id: 3, name: "JavaScript", link: "#" },
+      { id: 2, name: "CSS", link: "" },
+      { id: 3, name: "JavaScript", link: "../database/Roadmap/Js" },
```
Comment (dhairyagothi) [inline src/components/Roadmap/index.tsx:18]: It is not working Please tell changes to do

Comment (Ajay-Dhangar) [reply src/components/Roadmap/index.tsx:18]: 1. Navigate to the `pages/roadmap` directory and create a new file named `javascript/index.tsx`.
2. Import the content from `database/Roadmap/Js` into this new file.
3. Write the code for the new JavaScript roadmap page in `javascript/index.tsx`.
4. Update the import path in `src/components/Roadmap/index.tsx` by replacing `../database/Roadmap/Js` with `/roadmap/javascript`.

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
@dhairyagothi, change your code based on suggestions

**Comment by dhairyagothi** [comment]
image

added @Ajay-Dhangar

**Comment by Ajay-Dhangar** [comment]
@dhairyagothi, roadmap design is not good. we need good UI/UX for better understanding you can follow https://roadmap.sh/javascript

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
@dhairyagothi, Please focus on best practices when using CSS, especially since we are handling large amounts of content. It's important to use specific components and CSS rules, such as parent-child relationships, to ensure our styles are efficient and maintainable. Additionally, consider using tools like styled-components for better organization and scalability.

**Comment by dhairyagothi** [comment]
sure i am trying to make it as like you said

**Comment by dhairyagothi** [comment]
@Ajay-Dhangar i want to ask that if i can make a figma design and later convert it into html css and adding it .tsx file then its okay ?

**Comment by Ajay-Dhangar** [comment]
@dhairyagothi, I appreciate your willingness to help with my open source projects. However, please keep in mind that I want to respect your time, knowing you're actively contributing to multiple projects. Longer discussions might complicate things or create issues based on your PRs or contributions.

**Comment by dhairyagothi** [comment]
Sure @Ajay-Dhangar i will make pr ASAP

**Comment by dhairyagothi** [comment]
image

@Ajay-Dhangar please consider this for now

File: src/pages/roadmap/javascript/index.css [hunk 2]
```diff
@@ -0,0 +1,96 @@
+body {
```
Comment (Ajay-Dhangar) [inline src/pages/roadmap/javascript/index.css:1]: remove

```css
body {
    font-family: sans-serif;
    background-color: #f4f4f4;
  }
```

Comment (Ajay-Dhangar) [reply src/pages/roadmap/javascript/index.css:1]: you need to read carefully https://github.com/CodeHarborHub/codeharborhub.github.io/pull/2933#pullrequestreview-2172510159 I shared alreade for css

Comment (Ajay-Dhangar) [reply src/pages/roadmap/javascript/index.css:1]: for example you can check ou https://github.com/CodeHarborHub/codeharborhub.github.io/blob/main/src/components/Roadmap/Roadmap.css like this

Comment (Ajay-Dhangar) [reply src/pages/roadmap/javascript/index.css:1]: for example you can check ou https://github.com/CodeHarborHub/codeharborhub.github.io/blob/main/src/components/Roadmap/Roadmap.css like this

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
Please update the code based on the suggestions. Focus on enhancing the design and making data handling with props and mapping more efficient. For reference, you can visit https://github.com/CodeHarborHub/codeharborhub.github.io/tree/main/src/data/roadmaps. Create a JavaScript file there and add the relevant topics.

> NOTE: you can write code only one file like `javascript/index.tsx` 


### For Example

`src/database/roadmap/css/index.tsx`

```tsx
type ResourceLink = {
  url: string;
  text: string;
};

type SubTopic = {
  id: string;
  title: string;
  description: string;
  links: ResourceLink[];
};

type Topic = {
  id: number;
  title: string;
  content: SubTopic[];
};

const cssRoadmap: Topic[] = [
  {
    id: 1,
    title: "Introduction to CSS",
    content: [
      {
        id: "1.1",
        title: "What is CSS?",
        description: "Learn about the basics of CSS and its importance in web development.",
        links: [
          { url: "https://developer.mozilla.org/en-US/docs/Learn/CSS/First_steps/What_is_CSS", text: "MDN Web Docs" },
          { url: "https://www.w3schools.com/css/css_intro.asp", text: "W3Schools Introduction to CSS" }
        ]
      },
      {
        id: "1.2",
        title: "CSS Syntax and Selectors",
        description: "Understand the syntax of CSS and how to use selectors to style elements.",
        links: [
          { url: "https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors", text: "MDN CSS Selectors" },
          { url: "https://www.w3schools.com/css/css_syntax.asp", text: "W3Schools CSS Syntax" }
        ]
      }
    ]
  },
  {
    id: 2,
    title: "CSS Fundamentals",
    content: [
      {
        id: "2.1",
        title: "Colors and Backgrounds",
        description: "Learn how to apply colors and background styles to elements.",
        links: [
          { url: "https://developer.mozilla.org/en-US/docs/Web/CSS/color", text: "MDN Colors" },
          { url: "https://www.w3schools.com/css/css_background.asp", text: "W3Schools Backgrounds" }
        ]
      },
      {
        id: "2.2",
        title: "Box Model",
        description: "Understand the CSS box model and how it affects element layout.",
        links: [
          { url: "https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Box_Model/Introduction_to_the_CSS_box_model", text: "MDN Box Model" },
          { url: "https://www.w3schools.com/css/css_boxmodel.asp", text: "W3Schools Box Model" }
        ]
      }
    ]
  },
  {
    id: 3,
    title: "Advanced CSS",
    content: [
      {
        id: "3.1",
        title: "Flexbox",
        description: "Learn about the Flexbox layout model and how to use it for responsive designs.",
        links: [
          { url: "https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout", text: "MDN Flexbox" },
          { url: "https://www.w3schools.com/css/css3_flexbox.asp", text: "W3Schools Flexbox" }
        ]
      },
      {
        id: "3.2",
        title: "Grid",
        description: "Understand CSS Grid and how it can be used to create complex layouts.",
        links: [
          { url: "https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout", text: "MDN Grid" },
          { url: "https://www.w3schools.com/css/css_grid.asp", text: "W3Schools Grid" }
        ]
      }
    ]
  }
];

export default cssRoadmap;
```

**Pull request closed without merging**